### DISCUSSION TITLE
I want to draw a discharge capacity fade curve.

### DISCUSSION THREAD

**Discussion by YouHyin** [pybamm-team/PyBaMM#2415] [Q&A]
![capacity fade curve](https://user-images.githubusercontent.com/113442395/198958711-c04b69d5-c2d6-494b-b165-68aa754fab6a.png)

I must draw a discharge capacity fade curve like this picture to determine the battery cycle life.

[Code]
import pybamm
import matplotlib.pyplot as plt
import numpy as np

parameter_values = pybamm.ParameterValues("Mohtat2020")
parameter_values.update({"SEI kinetic rate constant [m.s-1]": 1e-14})
spm = pybamm.lithium_ion.SPM({"SEI": "ec reaction limited"})

param = spm.param
esoh_solver = pybamm.lithium_ion.ElectrodeSOHSolver(parameter_values, param)

Vmin = 3.0
Vmax = 4.2
Cn = parameter_values.evaluate(param.n.cap_init)
Cp = parameter_values.evaluate(param.p.cap_init)
n_Li_init = parameter_values.evaluate(param.n_Li_particles_init)

inputs={ "V_min": Vmin, "V_max": Vmax, "C_n": Cn, "C_p": Cp, "n_Li": n_Li_init}
esoh_sol = esoh_solver.solve(inputs)

print(f"Initial negative electrode SOC: {esoh_sol['x_100'].data[0]:.3f}")
print(f"Initial positive electrode SOC: {esoh_sol['y_100'].data[0]:.3f}")

# Update parameter values with initial conditions
c_n_max = parameter_values.evaluate(param.n.prim.c_max)
c_p_max = parameter_values.evaluate(param.p.prim.c_max)
parameter_values.update(
    {
        "Initial concentration in negative electrode [mol.m-3]": esoh_sol["x_100"].data[0] * c_n_max,
        "Initial concentration in positive electrode [mol.m-3]": esoh_sol["y_100"].data[0] * c_p_max,
    }
)

pybamm.set_logging_level("NOTICE")

experiment = pybamm.Experiment([
    (f"Discharge at 1C until {Vmin}V",
     "Rest for 1 hour",
     f"Charge at 1C until {Vmax}V",
     f"Hold at {Vmax}V until C/50"
    )
])
sim = pybamm.Simulation(spm, experiment=experiment, parameter_values=parameter_values)
sol = sim.solve()

experiment = pybamm.Experiment([
    (f"Discharge at 1C until {Vmin}V",
     "Rest for 1 hour",
    f"Charge at 1C until {Vmax}V",
    f"Hold at {Vmax}V until C/50")
] *_1000_,
termination="80% capacity"
)
sim = pybamm.Simulation(spm, experiment=experiment, parameter_values=parameter_values)
sol = sim.solve()

a1=sol["Discharge capacity [A.h]"].data
a2=sol.summary_variables["Cycle number"]
print(a2)

There's only 50 cycles here. What should I do to get the whole cycle (1000 I set up)?
I'd appreciate it if you could help me.

**Comment by YouHyin** [comment]
**Result : code**

![50](https://user-images.githubusercontent.com/113442395/198962402-470a7c25-f3b1-4ff5-b53b-f4bd9f42cc3b.png)

**Reply by rtimms** [reply]
It stopped because you dropped below 80% capacity (your termination condition).

**Reply by YouHyin** [reply]
Thank you. This has helped me a lot.

1. termination="80% capacity" : I meant to charge the SOC to 80% when charging the battery. Then what should I do if I want to charge it until it reaches 80% SOC?

2. I want to draw a discharge capacity fade curve like the picture I uploaded.
![capacity fade curve](https://user-images.githubusercontent.com/113442395/199177938-b76a3c95-ba17-4e88-ad9c-71f423c3637b.png)
What I can do?

In other words, I want to put the y-axis as the discharge capacity and the x-axis as the cycle number.
But only graphs with capacity (not discharge capacity in the y-axis) and cycle number in the x-axis could be obtained.
![capacity fade](https://user-images.githubusercontent.com/113442395/199182488-b10a19d3-ed57-4e49-8aa1-dda97510804c.png)

Please help me. I'd appreciate it if you could help me.

**Reply by rtimms** [reply]
1. You can't specify an SOC target in experiments yet. Try searching the discussions for "SOC" as there are lots of related posts.
2. You can get the capacity as a function of cycle number by doing `solution.summary_variables["Capacity [A.h]"]`. This will return an array so you can make your own plot.

**Reply by Notslytherin1** [reply]
i tried to do the same, but i noticed that the starting point of the capacity is very low (1.8) and normally the actual capacity is 63A.h, even if i optimized the parameters with experimental data.
Can you please give me an explanation
![capacity fade](https://github.com/user-attachments/assets/74edc94f-c021-40fc-81a6-00dbea3fb036)
![curvefitting](https://github.com/user-attachments/assets/a58767c8-6b2b-4f64-adc5-916f78db596e)

**Comment by Akila1993** [comment]
Hi @rtimms, 

I think the question is how to plot the Discharge capacity [A.h] against cycle number, not just the capacity. 

I also Have a similar question where I need to  plot Discharge capacity against the Total charge throughput capacity.  

but the variables only has throughout capacity and discharge capacity, so the if plot it gives a different plot than a continuous decaying discharge capacity. (Please refer below code snipped and figure) 

```python3
cycle_number = 10
exp = pybamm.Experiment(
    [
        "Hold at 4.2 V until C/100 (5 minute period)",
        "Rest for 4 hours (5 minute period)",
        "Discharge at 0.1C until 2.5 V (5 minute period)",  # initial capacity check
        "Charge at 0.3C until 4.2 V (5 minute period)",
        "Hold at 4.2 V until C/100 (5 minute period)",
    ]
    + [
        (
            "Discharge at 1C until 2.5 V",  # ageing cycles
            "Charge at 0.3C until 4.2 V (5 minute period)",
            "Hold at 4.2 V until C/100 (5 minute period)",
        )
    ]
    * cycle_number
    + ["Discharge at 0.1C until 2.5 V (5 minute period)"],  # final capacity check
)
sim = pybamm.Simulation(model, parameter_values=updated_parameters, experiment=exp, var_pts=var_pts)
sol = sim.solve()

Qt = sol["Throughput capacity [A.h]"].entries
DC = sol["Discharge capacity [A.h]"].entries
plt.plot(Qt, DC, label="DC")
```

![f9c271effc6a9620b43104fc495eb43811b6e1a1](https://github.com/user-attachments/assets/de71b7f1-3b90-4a5b-ae0d-692e22d2f0a9)
